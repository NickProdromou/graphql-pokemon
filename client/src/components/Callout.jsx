import React from "react";
import PropTypes from "prop-types";
import Styled from "styled-components";
import { Page, Row, Column } from "hedron";
import { spacing, colours } from "../style/variables";
import { type } from "../style/mixins/index";

const Callout = ({ title, subtitle }) => (
  <StyledCallout className="Callout">
    <Page>
      <Column fluid>
        <Row>
          <div className="CalloutInner">
            <h2 className="CalloutTitle">{title}</h2>
            <h3 className="CalloutSubtitle">{subtitle}</h3>
          </div>
        </Row>
      </Column>
    </Page>
  </StyledCallout>
);

Callout.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

Callout.defaultProps = {
  subtitle: ""
};

const StyledCallout = Styled.section`
  background: ${colours.primaryColorLight};
  padding: ${spacing.mid.level4} 0;

  .CalloutInner {
    padding: 0 ${spacing.mid.level1};
  }

  .CalloutTitle {
    ${type("heading2")}
  }

  .CalloutSubtitle {
    ${type("heading3")}
  }
`;

export default Callout;
