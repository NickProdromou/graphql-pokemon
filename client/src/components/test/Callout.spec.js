import React from "react";
import renderer from "react-test-renderer";
import Callout from "../Callout";

describe("Callout component", () => {
  const props = {
    title: "This is a title",
    subtitle: "this is a subtitle"
  };

  test("renders correctly with correct props", () => {
    const component = renderer.create(
      <Callout title={props.title} subtitle={props.subtitle} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("uses default props", () => {
    const component = renderer.create(
      <Callout title={props.title} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
