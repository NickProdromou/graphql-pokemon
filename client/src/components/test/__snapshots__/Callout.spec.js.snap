// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout component renders correctly with correct props 1`] = `
<section
  className="Callout sc-ifAKCX dVpvRL"
>
  <div
    className="sc-bxivhb bIAnAi"
  >
    <div
      className="sc-htpNat eWAgNU"
    >
      <section
        className="sc-bwzfXH irrVft"
      >
        <div
          className="CalloutInner"
        >
          <h2
            className="CalloutTitle"
          >
            This is a title
          </h2>
          <h3
            className="CalloutSubtitle"
          >
            this is a subtitle
          </h3>
        </div>
      </section>
    </div>
  </div>
</section>
`;

exports[`Callout component uses default props 1`] = `
<section
  className="Callout sc-ifAKCX dVpvRL"
>
  <div
    className="sc-bxivhb bIAnAi"
  >
    <div
      className="sc-htpNat eWAgNU"
    >
      <section
        className="sc-bwzfXH irrVft"
      >
        <div
          className="CalloutInner"
        >
          <h2
            className="CalloutTitle"
          >
            This is a title
          </h2>
          <h3
            className="CalloutSubtitle"
          >
            
          </h3>
        </div>
      </section>
    </div>
  </div>
</section>
`;
